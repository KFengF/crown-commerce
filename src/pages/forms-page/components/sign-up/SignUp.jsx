import React, { useState } from 'react';
import { connect } from 'react-redux'
import FormInput from '../form-input/FormInput';
import CustomButton from '../../../../components/custom-button/CustomButton';
import { signUpStart } from '../../../../utils/redux/user/userActions'
import { SignUpDiv, SignUpTitleH2, SignUpSpan } from './SignUpStyles';
/* import './SignUp.scss'; */

const SignUp = ({ signUpStart }) => {
    const initialState = {
        displayName: '',
        email: '',
        password: '',
        confirmPassword: ''
    }

    const [credentials, setCredentials] = useState(initialState);
    const { displayName, email, password, confirmPassword } = credentials;

    const onSubmit = async event => {
        event.preventDefault();

        if ( password !== confirmPassword ) {
            alert('passwords dont match');
            return;
        }

        try {
            await signUpStart({ displayName, email, password });
            setCredentials(initialState);
        } catch (error) {
            console.error(error);
        }
    }

    const onChange = event => {
        const { name, value } = event.target;

        setCredentials({ ...credentials, [name]: value }); 
        //los parentesis significa que lo que sea name sera el key
    }

    return (
        <SignUpDiv>
            <SignUpTitleH2>I do not have an account</SignUpTitleH2>
            <SignUpSpan>Sign up with your email and password</SignUpSpan>
            <form onSubmit={ onSubmit } >
                <FormInput 
                    type="text"
                    name="displayName"
                    value={ displayName }
                    onChange={ onChange }
                    label="Display Name"
                    required 
                />
                <FormInput 
                    type="email"
                    name="email"
                    value={ email }
                    onChange={ onChange }
                    label="Email"
                    required 
                />
                <FormInput 
                    type="password"
                    name="password"
                    value={ password }
                    onChange={ onChange }
                    label="Password"
                    required 
                />
                <FormInput 
                    type="password"
                    name="confirmPassword"
                    value={ confirmPassword }
                    onChange={ onChange }
                    label="Confirm Password"
                    required 
                />
                <CustomButton type="submit" >SIGN UP</CustomButton>
            </form>
        </SignUpDiv>
    );
}

const mapDispatch = dispatch => ({
    signUpStart: datas => dispatch(signUpStart(datas))
});

export default connect(null, mapDispatch)(SignUp);