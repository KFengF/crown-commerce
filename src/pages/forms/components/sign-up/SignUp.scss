.sign-up {
    width: 400px;
    display: flex;
    flex-direction: column;
    margin-left: 5%;

    .sign-up-title {
        margin-bottom: 10px;
    }

    span {
        letter-spacing: 1px;
    }
}