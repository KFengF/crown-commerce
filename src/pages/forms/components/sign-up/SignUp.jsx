import React from 'react';
import { auth, createUserDoc } from '../../../../utils/firebase';
import FormInput from '../form-input/FormInput';
import CustomButton from '../../../../components/custom-button/CustomButton';
import './SignUp.scss';

const initialState = {
    displayName: '',
    email: '',
    password: '',
    confirmPassword: ''
}

class SignUp extends React.Component {
    constructor () {
        super ();
        
        this.state = initialState
    }

    onSubmit = async event => {
        event.preventDefault();

        const { displayName, email, password, confirmPassword } = this.state;

        if ( password !== confirmPassword ) {
            alert('passwords dont match');
            return;
        }

        try {
            const { user } = await auth.createUserWithEmailAndPassword(email, password);

            await createUserDoc(user, { displayName });
            //No se porque se esta pasando el displayName
            this.setState(initialState)
            //Tampoco se porque se esta poniendo el estado inicial
        } catch (error) {
            console.error(error)
        }
    }

    onChange = event => {
        const { name, value } = event.target;

        this.setState({ [name]: value }); 
        //los parentesis significa que lo que sea name sera el key
    }

    render () {
        const { displayName, email, password, confirmPassword } = this.state;
        return (
            <div className="sign-up" >
                <h2 className="title" >I do not have an account</h2>
                <span>Sign up with your email and password</span>
                <form className="sign-up-form" onSubmit={ this.onSubmit } >
                    <FormInput 
                        type="text"
                        name="displayName"
                        value={ displayName }
                        onChange={ this.onChange }
                        label="Display Name"
                        required 
                    />
                    <FormInput 
                        type="email"
                        name="email"
                        value={ email }
                        onChange={ this.onChange }
                        label="Email"
                        required 
                    />
                    <FormInput 
                        type="password"
                        name="password"
                        value={ password }
                        onChange={ this.onChange }
                        label="Password"
                        required 
                    />
                    <FormInput 
                        type="password"
                        name="confirmPassword"
                        value={ confirmPassword }
                        onChange={ this.onChange }
                        label="Confirm Password"
                        required 
                    />
                    <CustomButton type="submit" >SIGN UP</CustomButton>
                </form>
            </div>
        );
    }
}

export default SignUp;