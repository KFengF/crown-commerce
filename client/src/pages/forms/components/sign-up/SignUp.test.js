import React from 'react';
import { mount } from 'enzyme';
import { SignUp } from './SignUp.component';

describe('SignUp component', () => {
    let wrapper;
    let mockSignUpStart;
    const originalError = console.error;
    //Usar esto para evitar los errores

    beforeEach(() => {
        mockSignUpStart = jest.fn();
        const mockProps = {
            signUpStart: mockSignUpStart
        }
        wrapper = mount(<SignUp { ...mockProps } />);
        console.error = () => {};
    });

    afterAll(() => {
        console.error = originalError;
    });

    it('should render SignUp component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call signUpStart when submit', done => {
        wrapper.find('input').at(0).simulate('change', 
            { target: { name: 'displayName', value: 'test jest' } }
        );
        wrapper.find('input').at(1).simulate('change', 
            { target: { name: 'email', value: 'testing@example.com' } }
        );
        wrapper.find('input').at(2).simulate('change', 
            { target: { name: 'password', value: '123456' } }
        );
        wrapper.find('input').at(3).simulate('change', 
            { target: { name: 'confirmPassword', value: '123456'} }
        );
        wrapper.find('form').simulate('submit', { preventDefault: () => {} });
        expect(mockSignUpStart).toHaveBeenCalled();
        done();
    });
});