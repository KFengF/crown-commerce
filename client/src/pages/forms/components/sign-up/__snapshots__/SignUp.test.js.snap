// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component should render SignUp component 1`] = `ReactWrapper {}`;
